src/tools.c

```diff
@@ -228,64 +228,125 @@ struct particle tools_init_orbit3d(double M, double m, double a, double e, doubl
 	return p;
 }
 
-#define TINY 1.0e-12
-struct orbit tools_p2orbit(struct particle p, struct particle star){
+struct orbit tools_orbit_nan(void){
```

Comment:
Instead of a function I would do this:

    const struct orbit orbit_nan = {.a = NAN,...}